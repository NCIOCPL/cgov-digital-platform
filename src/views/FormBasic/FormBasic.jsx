import React from 'react';
import PropTypes from 'prop-types';

//modules
import {
  Age,
  CancerTypeKeyword,
  ZipCode,
} from '../../components/search-modules/';

const FormBasic = () => {
  const handleSubmit = e => {
    e.preventDefault();
    console.log('form submitted');
  };

  return (
    <form onSubmit={handleSubmit} className="search-page__form advanced">
      <CancerTypeKeyword />
      <Age />
      <ZipCode />
    </form>
  );
};

FormBasic.propTypes = {
};

FormBasic.defaultProps = {
};

export default FormBasic;
