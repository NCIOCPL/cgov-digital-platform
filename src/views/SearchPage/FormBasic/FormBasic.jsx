import React, { useState } from 'react';
import PropTypes from 'prop-types';

//modules
import {
  Age,
  CancerTypeKeyword,
  ZipCode,
} from '../../../components/search-modules/';

const FormBasic = ({ handleUpdate }) => {
  const modules = [CancerTypeKeyword, [Age, ZipCode]];

  return (
    <>
      {modules.map((Module, idx) => {
        if (Array.isArray(Module)) {
          return (
            <div key={`formAdvanced-${idx}`} className="side-by-side">
              {Module.map((Mod, i) => (
                <Mod
                  key={`formAdvanced-${idx}-${i}`}
                  handleUpdate={handleUpdate}
                />
              ))}
            </div>
          );
        } else {
          return (
            <Module key={`formAdvanced-${idx}`} handleUpdate={handleUpdate} />
          );
        }
      })}
    </>
  );
};

FormBasic.propTypes = {
  handleUpdate: PropTypes.func,
};

export default FormBasic;
