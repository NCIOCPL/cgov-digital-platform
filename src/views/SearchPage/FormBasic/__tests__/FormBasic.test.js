import React from 'react';
import { render} from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import FormBasic from '../FormBasic';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <FormBasic {...props} />
  );
  return component;
};

describe('FormBasic', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
