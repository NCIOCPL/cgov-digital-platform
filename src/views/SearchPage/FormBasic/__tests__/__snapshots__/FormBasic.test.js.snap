// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBasic Render renders without error 1`] = `
Array [
  <fieldset
    class="cts-fieldset "
    id="fieldset--type"
  >
    <legend
      class="cts-fieldset__legend"
    >
      <span>
        Cancer Type/Keyword
      </span>
    </legend>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
    <div
      class="cts-fieldset__body"
    >
      <div
        class=""
      >
        <input
          aria-label="Cancer Type/Keyword"
          aria-required="false"
          class="cts-input null"
          id="q"
          name="CancerTypeKeyword"
          placeholder="Start typing to select a cancer type"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </fieldset>,
  <div
    class="side-by-side"
  >
    <fieldset
      class="cts-fieldset "
      id="fieldset--age"
    >
      <legend
        class="cts-fieldset__legend"
      >
        <span>
          Age
        </span>
      </legend>
      <a
        aria-label="Help"
        class="text-icon-help"
        href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
        rel="noopener noreferrer"
        target="_blank"
      >
        ?
      </a>
      <div
        class="cts-fieldset__body"
      >
        <div
          class=""
        >
          <label
            class="cts-label"
            for="a"
            id="a-label"
          >
            Enter the age of the participant.
          </label>
          <input
            aria-labelledby="a-label"
            aria-required="false"
            class="cts-input null"
            id="a"
            maxlength="3"
            name="age"
            spellcheck="false"
            type="text"
            value=""
          />
        </div>
      </div>
    </fieldset>
    <fieldset
      class="cts-fieldset "
      id="fieldset--zip"
    >
      <legend
        class="cts-fieldset__legend"
      >
        <span>
          U.S. Zip Code
        </span>
      </legend>
      <a
        aria-label="Help"
        class="text-icon-help"
        href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#basicsearch"
        rel="noopener noreferrer"
        target="_blank"
      >
        ?
      </a>
      <div
        class="cts-fieldset__body"
      >
        <div
          class=""
        >
          <label
            class="cts-label"
            for="z"
            id="z-label"
          />
          <input
            aria-labelledby="z-label"
            aria-required="false"
            class="cts-input null"
            id="z"
            maxlength="5"
            name="zip"
            spellcheck="false"
            type="text"
            value=""
          />
          <span
            class="cts-input__help-text"
          >
            Show trials near this U.S. ZIP code.
          </span>
        </div>
      </div>
    </fieldset>
  </div>,
]
`;
